src/Bicep.RegistryModuleTool/Extensions/TempFile.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO.Abstractions;
+
+namespace Bicep.RegistryModuleTool.Extensions
+{
+    internal sealed class TempFile : IDisposable
+    {
+        private readonly IFileSystem fileSystem;
+
+        private string? path;
+
+        public TempFile(IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+            this.path = fileSystem.Path.GetTempFileName();
+        }
+
+        public string Path => this.path ?? throw new ObjectDisposedException(this.GetType().Name);
+
+        ~TempFile() => this.DisposeInternal();
```

Comment:
>~ [](http://example.com/codeflow?start=8&length=1)

Do you need the finalizer if we don't have any native resources to clean up?

Reply:
The goal is to delete the file even if someone forgot to call `Dispose`, but I guess it's a bit of over-designed. Anyway, the file was removed in favor of `DeleteOnClose`.